src/box/lua/config/source/env.lua

```diff
@@ -75,6 +104,30 @@ end
 function methods.sync(self, _config_module, _iconfig)
     local values = {}
 
+    -- Handle old TT_* box.cfg()'s environment variables such as
+    -- IPROTO_LISTEN.
+    --
+    -- TODO: Here we miss all the options without 'box_cfg'
+    -- annotation: at least 'log', 'audit_log' and 'metrics'.
+    if self.name == 'env (default)' then
+        for _, w in instance_config:pairs() do
+            if w.schema.box_cfg ~= nil then
+                local value = box.internal.cfg.env[w.schema.box_cfg]
+                if value ~= nil then
+                    -- TT_LISTEN and TT_REPLICATION needs a
+                    -- special handling, because config's schema
+                    -- is more strict regarding a form an URI.
+                    if w.schema.box_cfg == 'listen' then
+                        value = normalize_uri_list(value)
+                    elseif w.schema.box_cfg == 'replication' then
+                        value = normalize_uri_list(value)
```

Comment:
That's incorrect. `replication.peers` is an array of strings. However, we possibly will update it to a list of instance names (in a separate PR).

Reply:
I'm going to eliminate `TT_REPLICATION` logic at all due to #9541.

Reply:
Updated.